### ISSUE TITLE
Support Octopus Deploy Version Controlled projects

### ISSUE THREAD

**Issue by roketworks** [nuke-build/nuke#910]
### Description

When trying to create a release in a version-controlled project the argument `--gitRef` is required. There is also an additional optional `--gitCommit` argument. https://octopus.com/docs/octopus-rest-api/octopus-cli/create-release#Creatingreleases-version-controlled

These arguments are not provided in the CreateReleaseTask so trying to create a release against a version-controlled project results in the following error: 
```
No gitRef value provided. Please provide the --gitRef argument passing the branch name or commit that contains the project details for this release.
Exit code: -1
ProcessException: Process 'octo.exe' exited with code -1.
   > C:\Users\x\.nuget\packages\octopustools\9.0.0\tools\octo.exe create-release --project="x" --version=1.0.0 --server=https://x.octopus.app --apiKey=[hidden]
   @ C:\Users\x\RiderProjects\x

   at Nuke.Common.Tooling.ProcessExtensions.AssertZeroExitCode(IProcess process)
   at Nuke.Common.Tools.Octopus.OctopusTasks.OctopusCreateRelease(OctopusCreateReleaseSettings toolSettings)
   at Nuke.Common.Tools.Octopus.OctopusTasks.OctopusCreateRelease(Configure`1 configurator)
   at Build.<get_CreateOctopusRelease>b__17_1() in C:\Users\mike\RiderProjects\integration-agent\build\Build.cs:line 145
   at Nuke.Common.Execution.BuildExecutor.<>c.<Execute>b__4_2(Action x)
   at Nuke.Common.Utilities.Collections.EnumerableExtensions.ForEach[T](IEnumerable`1 enumerable, Action`1 action)
   at Nuke.Common.Execution.BuildExecutor.Execute(NukeBuild build, ExecutableTarget target, IReadOnlyCollection`1 previouslyExecutedTargets, Boolean failureMode)
```

### Usage Example

```code
OctopusTasks.OctopusCreateRelease(settings => settings 
   ...
   .SetGitRef("main")
   .SetGitCommit("492dc09")
   ...
```

### Alternative

Using the argument configurator

```
OctopusTasks.OctopusCreateRelease(settings => settings 
   ...
   .SetProcessArgumentConfigurator(a => a.Add("--gitRef=main"))
   .SetProcessArgumentConfigurator(a => a.Add("--gitCommitf=492dc09"))
   ...
```

**Comment by andrew-from-toronto** [comment]
Resolved with #1222

**Comment by matkoch** [comment]
Thanks @andrew-from-toronto !
